docs/source/developers/python/building.rst

```diff
@@ -0,0 +1,645 @@
+
+   .. tab-item:: Linux and macOS
+      :sync: linux-macos
+
+      On macOS, any modern XCode or Xcode Command Line Tools (``xcode-select --install``)
+      are sufficient.
+
+      On Linux, for this guide, we require a minimum of gcc or clang 9.
+      You can check your version by running
+
+      .. code-block::
+
+         $ gcc --version
+
+      If the system compiler is older than gcc 9, it can be set to a newer version
+      using the ``$CC`` and ``$CXX`` environment variables:
+
+      .. code-block::
+
+         $ export CC=gcc-9
+         $ export CXX=g++-9
+
+
+   .. tab-item:: Windows
+      :sync: wins
+
+      Building on Windows requires one of the following compilers to be
+      installed:
+
+      - `Build Tools for Visual Studio 2022 <https://aka.ms/vs/17/release/vs_BuildTools.exe>`_ or
+      - Visual Studio 2022
+
+      During the setup of Build Tools, ensure at least one Windows SDK
+      is selected.
+
+Environment setup
+=================
+
+First, start from a fresh clone of Apache Arrow:
+
+.. code-block::
+
+   $ git clone https://github.com/apache/arrow.git
+
+There are two supported ways to set up the build environment for PyArrow: using
+**Conda** to manage the dependencies or using **pip** with manual dependency
+management.
+
+Both methods are shown bellow for Linux and macOS. For Windows, only the
+Conda-based setup is currently documented, skipping some of the
+Linux/macOS-only packages.
+
+Note that in case you are not using conda on a Windows platform, Arrow C++
+libraries need to be bundled with ``pyarrow``. For extra information see the
+Windows tab under the :ref:`pyarrow_build_section` section.
+
+.. tab-set::
+   :sync-group: language
+
+   .. tab-item:: Linux and macOS using conda
+      :sync: linux-macos
+
+      Pull in the test data and setup the environment variables:
+
+      .. code-block::
+
+         $ pushd arrow
+         $ git submodule update --init
+         $ export PARQUET_TEST_DATA="${PWD}/cpp/submodules/parquet-testing/data"
+         $ export ARROW_TEST_DATA="${PWD}/testing/data"
+         $ popd
+
+      The `conda <https://conda.io/>`_ package manager allows installing build-time
+      dependencies for Arrow C++ and PyArrow as pre-built binaries, which can make
+      Arrow development easier and faster.
+
+      Let's create a conda environment with all the C++ build and Python dependencies
+      from conda-forge, targeting development for Python 3.13:
+
+      .. code-block::
+
+         $ conda create -y -n pyarrow-dev -c conda-forge \
+               --file arrow/ci/conda_env_unix.txt \
+               --file arrow/ci/conda_env_cpp.txt \
+               --file arrow/ci/conda_env_python.txt \
+               --file arrow/ci/conda_env_gandiva.txt \
+               compilers \
+               python=3.13 \
+               pandas
+
+      As of January 2019, the ``compilers`` package is needed on many Linux
+      distributions to use packages from conda-forge.
```

Comment:
6 years later we should check whether that's still the case :laughing: and probably update the date.

Reply:
Will look into it!

I did update the System Requirements section for Linux/macOS and Windows. I tested the new Windows setup and should be good. As for the Linux/macOS min compiler version I haven't tested, only updated the versions as per the info I got online.

Reply:
I've looked into this a bit and haven't found much:

- We're still using the `compilers` package in our CI: https://github.com/apache/arrow/blob/982d31f35fd2cfe87494698dae9ef67d3333658c/ci/docker/conda-cpp.dockerfile#L37
- I didn’t find any explicit mention in the conda-forge docs that it’s no longer needed (checked both the [main docs](https://conda-forge.org/docs/) and the [package page](https://anaconda.org/conda-forge/compilers)).

I could open a test PR that removes it, but I’m not sure it’s worth the time? What do you think, @raulcd ?

Reply:
I agree, probably not worth the time, thanks for checking @AlenkaF